components/create-bounty/token-amount/create-bounty-token-amount.tsx

```diff
@@ -0,0 +1,309 @@
+          formattedValue: total.toFixed()
+        })
+      }
+
+      if(distributions.type === 'total'){
+        const rewardValue = BigNumber(issueAmount?.value).minus(distributions.totalServiceFees) 
+        setRewardAmount({
+          value: rewardValue.toFixed(),
+          floatValue: rewardValue.toNumber(),
+          formattedValue: rewardValue.toFixed()
+        })
+      }
+    }
+  }, [distributions])
+
+  function handleIssueAmountOnValueChange(values: NumberFormatValues, type: 'reward' | 'total') {
+    const setType = type === 'reward' ? setRewardAmount : setIssueAmount
+    if (
+      needValueValidation &&
+      +values.floatValue > +currentToken?.currentValue
+    ) {
+      setType(ZeroNumberFormatValues);
+      setInputError(t("bounty:errors.exceeds-allowance"));
+    } else if (values.floatValue < 0) {
+      setType(ZeroNumberFormatValues);
+    } else if (
+      values.floatValue !== 0 &&
+      BigNumber(values.floatValue).isLessThan(BigNumber(currentToken?.minimum))
+    ) {
+      setType(values); 
+      setInputError(t("bounty:errors.exceeds-minimum-amount", {
+          amount: currentToken?.minimum,
+      }));
+    } else {
+      if(isFunders) debouncedDistributionsUpdater(values.value, type)
+      setType(values);
+      if (inputError) setInputError("");
+    }
+  }
+
+  function handleIssueAmountBlurChange(type: 'reward' | 'total') {
+    if (needValueValidation && tokenBalance?.lt(issueAmount.floatValue)) {
+      const setType = type === 'reward' ? setRewardAmount : setIssueAmount
+      setType({ formattedValue: tokenBalance.toFixed() });
+    }
+  }
+
+  function selectTokens() {
+    return (
+      <TokensDropdown
+        token={currentToken}
+        label={labelSelect}
+        tokens={customTokens}
+        userAddress={userAddress}
+        canAddToken={canAddCustomToken}
+        addToken={addToken}
+        setToken={setCurrentToken}
+        disabled={false}
+        defaultToken={defaultToken}
+        showCurrencyValue={false}
+        needsBalance={isFunding}
+        noLabel
+      />
+    );
+  }
+
+  function inputNumber() {
+    return (
+      <InputNumber
+        symbol={currentToken?.symbol}
+        classSymbol=""
+        thousandSeparator
+        value={issueAmount?.value}
+        placeholder="0"
+        allowNegative={false}
+        decimalScale={decimals}
+        onValueChange={(e) =>  e.value !== issueAmount.value && handleIssueAmountOnValueChange(e, "total")}
+        onBlur={() => handleIssueAmountBlurChange("total")}
+        error={!!inputError}
+        helperText={
+          <>{inputError && <p className="p-small my-2">{inputError}</p>}</>
+        }
+      />
+    );
+  }
+
+  function RenderBalance() {
+    if (isFunding && isFunders) return null;
+    return (
+      <RewardInformationBalanceView
+        amount={tokenBalance.toFixed()}
+        symbol={currentToken?.symbol}
+      />
+    );
+  }
+
+  function renderPrimaryToken() {
```

Comment:
I'd recommend you to move this function to another file and make it a real component, as we have seen that type of component inside a component can lead to re-rendering issues

Reply:
In this situation, the rendering flow is not impacted, but it is a screen that has to be refactored with due care and on another card. It was maintaining the existing flow and only adding the necessary logic and design